src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -291,9 +291,9 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
         });
 
         this.registerReducer("filterAzureSubscriptions", async (state) => {
-            await promptForAzureSubscriptionFilter(state);
-            await this.loadAllAzureServers(state);
-
+            if (await promptForAzureSubscriptionFilter(state)) {
+                await this.loadAllAzureServers(state);
+            }
```

Comment:
Maybe add a return statement in finally? What happens if the functions error out?

Reply:
Both functions that get called are fully wrapped in try-catches, but I'll add that here just in case something slips through.